hydropandas/extensions/plots.py

```diff
@@ -691,13 +714,37 @@ def series_per_group(self, plot_column, by=None, savefig=True, outputdir="."):
             ax.set_ylabel(row["unit"])
             ax.grid(True)
             f.tight_layout()
+
             if savefig:
-                if isinstance(row.obs, GroundwaterObs):
-                    name = name.split("-")[0]
+                if isinstance(by, list):
+                    by_name = "-".join(by)
+                else:
+                    by_name = by
+                    groupname = "-".join(groupname)
+                if naming_method is None:
+                    filename = f"series_by_{by_name}_group_{groupname}.png"
+                elif naming_method == "infer_name_monitoring_well":
+                    if "-" in name:
+                        filename = f"{name.split('-')[0]}.png"
+                    elif "_" in name:
+                        filename = f"{name.split('_')[0]}.png"
+                    else:
+                        filename = f"{name}.png"
+                elif isinstance(naming_method, str):
+                    filename = (
+                        f"series_by_{by_name}_group_"
+                        f"{'-'.join(group[naming_method].unique().tolist())}.png"
+                    )
+
                 f.savefig(
-                    os.path.join(outputdir, f"{name}.png"), bbox_inches="tight", dpi=150
+                    os.path.join(outputdir, filename), bbox_inches="tight", dpi=150
                 )
                 plt.close(f)
+            else:
+                axes.append(ax)
+
+        if not savefig:
+            return axes
```

Comment:
Include figure (`f`) as well.

Reply:
figures can be obtained from axes with ax.figure, e.g. `[iax.figure for iax in axes]`, so this should be okay.